core/auth/src/main/java/software/amazon/awssdk/auth/token/signer/SdkTokenExecutionAttribute.java

```diff
@@ -15,21 +15,73 @@
 
 package software.amazon.awssdk.auth.token.signer;
 
+import static software.amazon.awssdk.utils.CompletableFutureUtils.joinLikeSync;
+
+import java.util.concurrent.CompletableFuture;
 import software.amazon.awssdk.annotations.SdkProtectedApi;
 import software.amazon.awssdk.auth.token.credentials.SdkToken;
+import software.amazon.awssdk.core.SelectedAuthScheme;
 import software.amazon.awssdk.core.interceptor.ExecutionAttribute;
+import software.amazon.awssdk.core.interceptor.SdkInternalExecutionAttribute;
+import software.amazon.awssdk.http.auth.BearerAuthScheme;
+import software.amazon.awssdk.http.auth.BearerHttpSigner;
+import software.amazon.awssdk.http.auth.spi.AuthSchemeOption;
+import software.amazon.awssdk.identity.spi.Identity;
 
 /**
  * SdkToken authorizing attributes attached to the execution.
+ *
+ * @deprecated Signer execution attributes have been deprecated in favor of signer properties, set on the auth scheme's signer
+ * options.
  */
+@Deprecated
 @SdkProtectedApi
 public final class SdkTokenExecutionAttribute {
 
     /**
      * The token to sign requests using token authorization instead of AWS Credentials.
+     *
+     * @deprecated This is a protected class that is internal to the SDK, so you shouldn't be using it.
      */
-    public static final ExecutionAttribute<SdkToken> SDK_TOKEN = new ExecutionAttribute<>("SdkToken");
+    @Deprecated
+    public static final ExecutionAttribute<SdkToken> SDK_TOKEN =
+        ExecutionAttribute.derivedBuilder("SdkToken",
+                                          SdkToken.class,
+                                          SdkInternalExecutionAttribute.SELECTED_AUTH_SCHEME)
+                          .readMapping(SdkTokenExecutionAttribute::sdkTokenReadMapping)
+                          .writeMapping(SdkTokenExecutionAttribute::sdkTokenWriteMapping)
+                          .build();
 
     private SdkTokenExecutionAttribute() {
     }
+
+
+    private static SdkToken sdkTokenReadMapping(SelectedAuthScheme<?> authScheme) {
+        if (authScheme == null) {
+            return null;
+        }
+        Identity identity = joinLikeSync(authScheme.identity());
```

Comment:
There isn't currently a way to check if the type was an SDK token without joining, and building one felt overkill. Because this is a legacy code path and that code-path is already blocking, this didn't seem like a regression.

Reply:
You are right, since the existing identity resolution is blocking, it wouldn't be a regression. So this is fine, as yeah, building something to know the type for this would not be worth it.